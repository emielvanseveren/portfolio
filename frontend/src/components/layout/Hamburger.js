import React from 'react'
import styled from 'styled-components'

const Container = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  right: 100px;
  top: 100px;
  cursor: pointer;
`
const Line = styled.div`
  width: 15px;
  height: 2px;
  margin-bottom: 2px;
  background-color: ${({ inRange }) => inRange? 'white' : 'black'};
`

export default function Hamburger({inRange}){
  return(
    <Container>
      <Line inRange={inRange}/>
      <Line inRange={inRange}/>
      <Line inRange={inRange}/>
    </Container>
  )
}