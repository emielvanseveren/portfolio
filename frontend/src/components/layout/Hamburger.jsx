import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import size from '../../constants/size'

const Container = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  right: 100px;
  top: 100px;
  z-index: 3;
  cursor: pointer;

  @media ${size.xs}{
    top: 75px;
    right: 30px;
  }

`
const Line = styled.div`
  width: 15px;
  height: 2px;
  margin-bottom: 2px;
  background-color: ${({ inRange }) => inRange ? 'white' : 'black'};

  @media ${size.xs}{
    width: 20px;
    height: 3px;
    margin-bottom: 3px;
  }
`

export default function Hamburger({ inRange }){
  return (
    <Container>
      <Line inRange={inRange}/>
      <Line inRange={inRange}/>
      <Line inRange={inRange}/>
    </Container>
  )
}

Hamburger.propTypes = {
  inRange: PropTypes.bool.isRequired
}
