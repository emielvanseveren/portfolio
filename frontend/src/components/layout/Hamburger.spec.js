import React from 'react'
import Hamburger from './Hamburger'

// should add theme provider as this is created in a parent component
describe('<Hamburger/>', () => {
  it('Snapshot', () => {
    expect(mountWithRouterAndTheme(
      createComponent()
    )).toMatchSnapshot()
  })
})

function createComponent(props){
  const defaultprops = { inRange: false }
  return <Hamburger {...defaultprops} {...props}/>
}
