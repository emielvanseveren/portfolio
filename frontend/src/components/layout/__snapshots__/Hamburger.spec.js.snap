// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger/> Snapshot 1`] = `
<Hamburger
  inRange={false}
>
  <Hamburger__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Hamburger__Container-sc-1tzi5x8-0",
            "isStatic": false,
            "lastClassName": "jePrcg",
            "rules": Array [
              "position:absolute;display:flex;flex-direction:column;align-items:center;right:100px;top:100px;cursor:pointer;",
            ],
          },
          "displayName": "Hamburger__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Hamburger__Container-sc-1tzi5x8-0",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="Hamburger__Container-sc-1tzi5x8-0 jePrcg"
      >
        <Hamburger__Line
          inRange={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Hamburger__Line-sc-1tzi5x8-1",
                  "isStatic": false,
                  "lastClassName": "brCjHO",
                  "rules": Array [
                    "width:15px;height:2px;margin-bottom:2px;background-color:",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "Hamburger__Line",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Hamburger__Line-sc-1tzi5x8-1",
                "target": "div",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            inRange={false}
          >
            <div
              className="Hamburger__Line-sc-1tzi5x8-1 brCjHO"
            />
          </StyledComponent>
        </Hamburger__Line>
        <Hamburger__Line
          inRange={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Hamburger__Line-sc-1tzi5x8-1",
                  "isStatic": false,
                  "lastClassName": "brCjHO",
                  "rules": Array [
                    "width:15px;height:2px;margin-bottom:2px;background-color:",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "Hamburger__Line",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Hamburger__Line-sc-1tzi5x8-1",
                "target": "div",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            inRange={false}
          >
            <div
              className="Hamburger__Line-sc-1tzi5x8-1 brCjHO"
            />
          </StyledComponent>
        </Hamburger__Line>
        <Hamburger__Line
          inRange={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Hamburger__Line-sc-1tzi5x8-1",
                  "isStatic": false,
                  "lastClassName": "brCjHO",
                  "rules": Array [
                    "width:15px;height:2px;margin-bottom:2px;background-color:",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "Hamburger__Line",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Hamburger__Line-sc-1tzi5x8-1",
                "target": "div",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            inRange={false}
          >
            <div
              className="Hamburger__Line-sc-1tzi5x8-1 brCjHO"
            />
          </StyledComponent>
        </Hamburger__Line>
      </div>
    </StyledComponent>
  </Hamburger__Container>
</Hamburger>
`;
